### ISSUE TITLE
[NETSDKE2E]Workloads are not displayed in "dotnet workload list" after upgrade from NET RC2 to NET7 RTM.

### ISSUE THREAD

**Issue by vdanche** [dotnet/sdk#28451]
1. Install NET7 RC2 SDK.
2. Install below workloads with commands.
dotnet workload install android --skip-manifest-update
dotnet workload install wasm-tools --skip-manifest-update
3. check the workloads via "dotnet workload list"
![image](https://user-images.githubusercontent.com/65637796/195043530-be03f0b0-04d8-4f09-858f-b01b74735066.png)
4. Upgrade SDK to NET7 RTM SDK and then check the workloads again using "dotnet workload list"
**The workloads installed in NET7 RC2 are not displayed**
![image](https://user-images.githubusercontent.com/65637796/195044264-a7d3a72a-d1cc-4617-a83e-8d21fcd04c3f.png)

**Comment by marcpopMSFT** [comment]
Do we need a `dotnet workload list --all`? I think it's correctly pointing out that there are no workloads installed with the current band but there are some that could be updated. I think this is because we treat different previews as different feature bands and won't happen when going from 7.0.100 to 7.0.101 but we'll want to test that.

@vdanche what's the output of `dotnet workload --info`?

**Comment by marcpopMSFT** [comment]
Can you share the https://aka.ms/vscollect.exe logs? I wonder if the finalizer ran during upgrade and may have also cleaned up those workloads.

**Comment by vdanche** [comment]
@marcpopMSFT  dotnet workload --info as below
In NET7 RC2
![image](https://user-images.githubusercontent.com/65637796/195491543-fa4657c9-a924-4fe4-be8e-57274a9f1e17.png)

Upgrade to NET7 RTM
![image](https://user-images.githubusercontent.com/65637796/195491602-45d35d76-3fbf-4822-ba25-8e75ae0a36c8.png)


logs has been attached
[vslogs.zip](https://github.com/dotnet/sdk/files/9770179/vslogs.zip)

**Comment by marcpopMSFT** [comment]
@joeloff from the logs, it does look like the finalizer is kicking in and uninstalling the workloads. Is this a side effect of our upgrade path and finalizer?

**Comment by joeloff** [comment]
Yes, though even if the finalizer did not run, we'd still have a problem because the manifest feature bands are considered SxS. The finalizer is supposed to not run during the uninstall of an upgrade, but scheduling is difficult because the bundle doesn't expose enough information for us to use. I need to take a closer look at the logs to see what triggered it.

**Comment by marcpopMSFT** [comment]
We believe this is only a preview to preview issue. The only way prove that though since the preview workload logic went into 7.0 would be to install one stable build of 7.0.100, install workloads, and then upgrade to a later stable build of 7.0.100.  @vdanche is that something you might be able to try? not sure how you find two different stable versions of the 7.0.100.

If we prove that works, I think we're going to say that supporting this in previews is not a priority as it would be quite complex and fragile to do so.

**Comment by vdanche** [comment]
@marcpopMSFT Marc, currently 7.0.100 SDK version is always as below, could not find two different stable versions of the 7.0.100.
![image](https://user-images.githubusercontent.com/65637796/196142835-035dd647-3917-4b51-a10e-20ae4ab3cb04.png)
Addtionally, this issue also repro on NET6.0(6.0.401 to 6.0.402)

**Comment by marcpopMSFT** [comment]
@joeloff , can you check on the 401 to 402 upgrade as that shouldn't trigger the finalizer?

**Comment by joeloff** [comment]
Checked and it does trigger the finalizer - the finalizer always runs, but in the 401->402 case it should not be doing anything. I'm going to me this to dotnet/installer

**Comment by joeloff** [comment]
@dsplaisted I'm going to grab this issue and assign to myself since it's in the finalizer code.
